src/formatDate.js

```diff
@@ -50,7 +50,71 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldDateArr = date.split(fromFormat[3]);
+  let YYYY = 0;
+  let MM = 0;
+  let DD = 0;
+  let YY = 0;
+  const newDate = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        YYYY = oldDateArr[i];
+        break;
+
+      case 'YY':
+        YY = oldDateArr[i];
+        break;
+
+      case 'MM':
+        MM = oldDateArr[i];
+        break;
+
+      case 'DD':
+        DD = oldDateArr[i];
+        break;
+
+      default:
```

Comment:
Imagine if you have object (for ex. `parsedDate`) that stores your `YY`, `MM`, `DD`. So you can avoid switch/case and simple write:
```
parsedDate[fromFormat[i] = oldDateArr[i]
```